### PR TITLE
Implemented isNaN and so forth for inspecting floating point quantities

### LINKED ISSUE #134: Wrappers for isNan, isInfinite
What is the recommended way of validating that the floating point quantities someone gave you are actual numbers?

If you are receiving them as `Double`s and using `*~` to add on the unit, it seems clear that you can easily do the validation yourself before the multiplication.

However, it also seems very convenient to use the `Binary` or `Storable` or `FromJSON` instances for, e.g., `Velocity Double` to build up serialization and deserialization functions for entire records at once. (This becomes even more useful with the forthcoming fixed point support.) If you do that, then the unit ascriptions become implicit and you never see the `Double` before it becomes a `Quantity Double`.

Would it make sense to add wrappers like `isInfiniteQuantity :: RealFloat a => Quantity d a -> Bool` to perform this sort of validation after deserialization?

### PR DESCRIPTION
**Pull request by dmcclean**
Fixes #134.

### CODE REVIEW

File: src/Numeric/Units/Dimensional/Float.hs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import Numeric.Units.Dimensional.Prelude hiding (RealFloat(..))
+import Numeric.Units.Dimensional.Coercion
+
+-- | 'True' if the representation of the argument is too small to be represented in normalized format.
+isDenormalized :: RealFloat a => Quantity d a -> Bool
+isDenormalized = P.isDenormalized . unQuantity
+
+-- | 'True' if the representation of the argument is a number and is not infinite.
+isFiniteNumber :: RealFloat a => Quantity d a -> Bool
+isFiniteNumber = not . liftA2 (&&) isNaN isInfinite
```
Comment (bjornbm) [inline src/Numeric/Units/Dimensional/Float.hs]: `(||)`?

Comment (dmcclean) [inline src/Numeric/Units/Dimensional/Float.hs]: Yeah, I de-deMorganized that, apparently...

**Comment by dmcclean** [comment]
I didn't have the last PR #136 merged in with this when I was testing it, and it matters because I had added a documentation property in #136 that couldn't be tested. Fixing it now. There also may be some issue with GHCi printing one of the example error messages differently under travis's setup, I'm scratching my head on that one.

**Comment by dmcclean** [comment]
```
### Failure in /home/travis/build/dmcclean/dimensional/src/Numeric/Units/Dimensional.hs:105: expression `let x = 1 *~ meter + 1 *~ second'

expected: ...
              Couldn't match type 'Numeric.NumType.DK.Integers.Zero
                             with 'Numeric.NumType.DK.Integers.Pos1
              Expected type: Unit 'Metric DLength a
                Actual type: Unit 'Metric DTime a
              In the second argument of `(*~)', namely `second'
              In the second argument of `(+)', namely `1 *~ second'

 but got: 
          <interactive>:70:27:
              Couldn't match type ‘'Numeric.NumType.DK.Integers.Zero’
                             with ‘'Numeric.NumType.DK.Integers.Pos1’
              Expected type: Unit 'Metric DLength a
                Actual type: Unit 'Metric DTime a
              In the second argument of ‘(*~)’, namely ‘second’
              In the second argument of ‘(+)’, namely ‘1 *~ second’
```

The `...` is correctly matching the `<interactive>` bit with the not very predictable line number. But why are the types getting wrapped with extra curly single quotes? And why does this happen for the example on line 105, but the one on line 116 works? And why does it not exhibit this behavior on my machine, which as far as I can tell is running the same cabal and ghci versions.

**Comment by bjornbm** [comment]
I also see an “upgrade” to curly quotes on the last two lines.

**Comment by dmcclean** [comment]
Good point.

I have absolutely no idea what's going on here.

Tempted to just delete the examples of bad error messages and move them to the readme?

**Comment by dmcclean** [comment]
Like so.

**Comment by bjornbm** [comment]
I guess it isn't a battle worth fighting for now. Nice addition with the doctests test suite! :+1:

**Pull request merged**